internal/controllers/cluster/cluster_controller.go

```diff
@@ -331,6 +332,7 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 	}
 
 	controllerutil.RemoveFinalizer(cluster, clusterv1.ClusterFinalizer)
+	r.recorder.Eventf(cluster, corev1.EventTypeNormal, "Deleted", "Cluster %s has been deleted", cluster.Name)
```

Comment:
I think this event could be emitted multiple times as it depends on how quickly the garbage collector in kube-controller-manager actually removes the object.

But I assume we don't care about how often this event is emitted?

Reply:
Personally I think that's acceptable. From the CAPI perspective the action happened so it's appropriate for us to emit the event.

Reply:
I think this is acceptable too